src/services/utilServices/MailClient.ts

```diff
@@ -5,6 +5,13 @@ import { config } from "@config/config"
 import logger from "@logger/logger"
 
 const POSTMAN_API_URL = "https://api.postman.gov.sg/v1"
+const MAIL_VERIFICATION_DELAY = 60000 // 60 seconds
+
+type MailData = {
+  id: string
+  recipient: string
+  status: "UNSENT" | "ACCEPTED" | "SENT" | "BOUNCED" | "DELIVERED" | "OPENED"
```

Comment:
can abstract to const MailDataStatus = { Unsent: "UNSENT"...}

Reply:
can further abstract to something like:

```
const Roles = {
  Admin: "admin",
  Writer: "writer",
  Reader: "reader"
} as const;

// Convert object key in a type
type RoleKeys = typeof Roles[keyof typeof Roles]

declare function hasAccess(role: RoleKeys): void;
```
https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh

Reply:
done!